src/main/java/frc/robot/Robot.java

```diff
@@ -79,15 +89,19 @@
   private boolean m_ballDetectedExitLastLoop = false;
 
   // Launching
-  WPI_VictorSPX m_motorLauncherLeft =
+  private final WPI_VictorSPX m_motorLauncherLeft =
       new WPI_VictorSPX(RoboRIO.kPortMotorLauncherLeft);
-  WPI_TalonSRX m_motorLauncherRight =
+  private final WPI_TalonSRX m_motorLauncherRight =
       new WPI_TalonSRX(RoboRIO.kPortMotorLauncherRight);
+  private final DoubleSolenoid m_doubleSolenoidLauncherCannon =
+      new DoubleSolenoid(RoboRIO.kPortDoubleSolenoidForwardLauncherCannon,
+          RoboRIO.kPortDoubleSolenoidBackwardLauncherCannon);
   private final AnalogInput m_analogInputUltrasonicSensor =
       new AnalogInput(RoboRIO.kPortUltrasonicSensorPort);
   // Leave this uninitialized because we have to configure the analog input.
   private AnalogPotentiometer m_ultrasonicSensor;
   private boolean m_launchBall = false;
+  private boolean m_raiseLauncherCannon = false;
```

Comment:
This is a state variable, so it should probably be reset to `false` again in `disabledInit()` to prevent any weird behavior.